mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java

```diff
@@ -395,6 +406,42 @@ private void findAndMarkDisabledTaskExecutors(CheckDisabledTaskExecutors r) {
         }
     }
 
+    private void onDeleteDisabledTaskExecutorRequest(DeleteDisabledTaskExecutorRequest request) {
+        try {
+            mantisJobStore.deleteDisabledTaskExecutor(request.getTaskExecutorID());
+        } catch (IOException e) {
+            log.error("Error while deleting persisted disabled task executor {}", request.getTaskExecutorID(), e);
+        }
+        disabledTaskExecutors.remove(request.getTaskExecutorID());
+        sender().tell(Ack.getInstance(), self());
+    }
+
+    private void onDisableTaskExecutorRequest(DisableTaskExecutorRequest request) {
+        try {
+            log.info("Request to disable task executor {}", request.getTaskExecutorID());
+
+            final TaskExecutorState state = executorStateManager.get(request.getTaskExecutorID());
+            log.info("Trying to disable task executor {} with state {}", request.getTaskExecutorID(), state);
+
+            if (state == null) {
+                // If the TE is unknown by mantis, delete it from state
+                mantisJobStore.deleteDisabledTaskExecutor(request.getTaskExecutorID());
+                disabledTaskExecutors.remove(request.taskExecutorID);
+                sender().tell(new TaskExecutorOnDisabled(false), self());
+            } else {
```

Comment:
What if the task executor registers after this message gets processed? what should the semantics be in that case?

Reply:
It appears that we have different semantics for this than when the disabled request is for an ASG.